src/adapter/src/coord/sequencer/cluster.rs

```diff
@@ -950,6 +975,22 @@ impl Coordinator {
             );
         }
 
+        if disks.len() == 1 {
+            let disk = disks.into_iter().next().expect("must exist");
+            match &options.disk {
+                AlterOptionParameter::Set(ds) if *ds != disk => {
+                    coord_bail!(
+                        "Cluster replicas with DISK {disk} do not match expected DISK {ds}"
+                    );
+                }
+                _ => *new_disk = disk,
+            }
+        } else if !disks.is_empty() {
+            coord_bail!(
+                "Cannot convert unmanaged cluster to managed, non-unique replica DISK options"
```

Comment:
I do not think this condition is tested in a test:

```
CREATE CLUSTER c REPLICAS (r1 (SIZE '1', DISK), r2 (SIZE '1'));
ALTER CLUSTER c SET (managed=true);
```

Also, the error message is difficult to read , I think the comma must be a colon?